src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -576,6 +576,18 @@ private void Validate()
                 alias.Alias.CheckConstraints(semanticDiagnostics);
             }
 
+            foreach (var @using in Usings)
+            {
+                // Check if `using static` directives meet constraints.
+                if (@using.NamespaceOrType.IsType)
+                {
+                    var typeSymbol = (TypeSymbol)@using.NamespaceOrType;
+                    var corLibrary = typeSymbol.ContainingAssembly.CorLibrary;
```

Comment:
>var corLibrary = typeSymbol.ContainingAssembly.CorLibrary; [](start = 20, length = 58)

I think it is more appropriate to request CorLibrary from ```_compilation.SourceAssembly``` rather than from the assembly that defines the type. The type name in the using was resolved in context of the ```_compilation```. We will also be able to share a single ```TypeConversions``` instance to check all types.

Reply:
Addressed in 2fb9e18